src/SFA.DAS.EmployerIncentives.Commands/UpdateVrfCaseStatusForLegalEntity/UpdateVendorRegistrationCaseStatusForLegalEntityCommandHandler.cs

```diff
@@ -0,0 +1,29 @@
+﻿using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Commands.Persistence;
+using SFA.DAS.EmployerIncentives.Commands.Types.LegalEntity;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Commands.UpdateVrfCaseStatusForLegalEntity
+{
+    public class UpdateVendorRegistrationCaseStatusForLegalEntityCommandHandler : ICommandHandler<UpdateVendorRegistrationCaseStatusForLegalEntityCommand>
+    {
+        private readonly IAccountDomainRepository _domainRepository;
+
+        public UpdateVendorRegistrationCaseStatusForLegalEntityCommandHandler(IAccountDomainRepository domainRepository)
+        {
+            _domainRepository = domainRepository;
+        }
+
+        public async Task Handle(UpdateVendorRegistrationCaseStatusForLegalEntityCommand command, CancellationToken cancellationToken = default)
+        {
+            var accounts = await _domainRepository.GetByHashedLegalEntityId(command.HashedLegalEntityId);
+
+            foreach (var account in accounts)
```

Comment:
Why not just publish domain events for each individual account using the account.Id and HashedLegalEntityId?  Then this handler is an adapter to make a multi account update into a single account update.